### PR TITLE
[HOPSWORKS-3033] Python API for projects, datasets, jobs, executions and git

### PR DESCRIPTION
**Pull request by robzor92**

### CODE REVIEW

**Comment by robzor92** [comment]
Fixes #2

File: python/hopsworks/engine/git_engine.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            time.sleep(5)
+
+        if git_op.success is False:
+            raise GitException(
+                "Git command failed with the message: {}".format(
+                    git_op.command_result_message
+                )
+            )
+        else:
+            self._log.info("Git command {} finished".format(command))
```
Comment (gibchikafa) [inline python/hopsworks/engine/git_engine.py:60]: Here you can also log the `git_op.command_result_message`

Comment (robzor92) [reply python/hopsworks/engine/git_engine.py:60]: The API returns a List of GitFileStatus objects with the changes actually.

Comment (robzor92) [reply python/hopsworks/engine/git_engine.py:60]: For example [GitFileStatus('model_design_doc.md', 'D', '')]

**Pull request merged**